cmd/api/src/api/v2/cypher_search.go

```diff
@@ -30,14 +33,28 @@ type CypherSearch struct {
 }
 
 func (s Resources) CypherSearch(response http.ResponseWriter, request *http.Request) {
-	var payload CypherSearch
+	var (
+		payload CypherSearch
+		authCtx = ctx.FromRequest(request).AuthCtx
+	)
 
 	if err := api.ReadJSONRequestPayloadLimited(&payload, request); err != nil {
 		api.WriteErrorResponse(
 			request.Context(),
 			api.BuildErrorResponse(http.StatusBadRequest, "JSON malformed.", request), response,
 		)
-	} else if graphResponse, err := s.GraphQuery.RawCypherSearch(request.Context(), payload.Query, payload.IncludeProperties); err != nil {
+	} else if preparedQuery, err := s.GraphQuery.PrepareCypherQuery(payload.Query); err != nil {
+		api.WriteErrorResponse(
+			request.Context(),
+			api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response,
+		)
+	} else if preparedQuery.HasMutation && !s.Authorizer.AllowsPermission(authCtx, auth.Permissions().GraphDBMutate) {
+		s.Authorizer.AuditLogUnauthorizedAccess(request)
+		api.WriteErrorResponse(
+			request.Context(),
+			api.BuildErrorResponse(http.StatusForbidden, "Permission denied: User may not modify the graph.", request), response,
+		)
+	} else if graphResponse, err := s.GraphQuery.RawCypherSearch(request.Context(), preparedQuery, payload.IncludeProperties); err != nil {
```

Comment:
Is RawCypherQuery not just opening a read transaction by default?  If not, is that something we want to address?

Reply:
The definition of RawCypherSearch determines whether to use a Read or Write transaction based on preparedQuery.HasMutation. Previously, it would just pass directly to a Read transaction, but we now want to support both non-mutation and mutation queries